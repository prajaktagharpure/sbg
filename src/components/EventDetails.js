import React from 'react'
import MarketList from './MarketList'
import PropTypes from 'prop-types'
import U from '../utils/Util'

const EventDetails = ({ eventDetails, currentEventData, handleAddBetSlip }) => (
  <ul className='collection with-header event-details'>
    <li className='collection-header'>
      <h4>Event Details</h4>
    </li>
    <li className='collection-item'>
      <strong>Event Name </strong>
      {eventDetails.name}
    </li>
    <li className='collection-item'>
      <strong>Class Name </strong>
      {eventDetails.className}
    </li>
    <li className='collection-item'>
      <strong>Type Name</strong> {eventDetails.typeName}
    </li>
    <li className='collection-item'>
      <strong>StartTime </strong>
      {eventDetails.startTime}
    </li>
    <li className='collection-item'>
      <strong> Home </strong> {eventDetails.scores.home} <br />
      <strong> Away </strong>
      {eventDetails.scores.away}
    </li>
    <li className='collection-item'>
      <strong>Competitors Home</strong> {eventDetails.scores.home} <br />
      {eventDetails.competitors.map(competitor => competitor.name)}
    </li>
    {currentEventData.loading && (
      <>
        <i className='center-align loadingtext'>{currentEventData.loading}</i>
        <div className='progress indigo darken-4'>
          <div className='indeterminate' />
        </div>
      </>
    )}
    {!U.isObjEmpty(currentEventData) && currentEventData.marketData && (
      <MarketList
        ukey='ed'
        markets={currentEventData.marketData.slice(0, 19)}
        eventDetails={eventDetails}
        handleAddBetSlip={handleAddBetSlip}
      />
    )}
  </ul>
)
EventDetails.propTypes = {
  currentEventData: PropTypes.object.isRequired,
  eventDetails: PropTypes.object.isRequired,
  handleAddBetSlip: PropTypes.func
}
export default EventDetails
