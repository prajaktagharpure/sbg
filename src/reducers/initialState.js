export default {
  events: [],
  market: [],
  eventFullDetails: [],
  betSlips: []
}
