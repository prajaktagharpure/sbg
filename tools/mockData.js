const sportEvents = [
  {
    eventId: 21249939,
    name: 'Shanghai Shenhua 0 v 0 Shandong Luneng Taishan',
    displayOrder: -1000,
    sort: 'MTCH',
    linkedEventId: 21228740,
    classId: 5,
    className: 'Football',
    typeId: 10003971,
    typeName: 'Football Live',
    linkedEventTypeId: 10005942,
    linkedEventTypeName: 'Chinese Super League',
    startTime: '2017-09-19T11:35:23.000Z',
    scores: { home: 0, away: 0 },
    competitors: [
      { name: 'Shanghai Shenhua', position: 'home' },
      { name: 'Shandong Luneng Taishan', position: 'away' }
    ],
    status: {
      active: true,
      started: true,
      live: true,
      resulted: false,
      finished: false,
      cashoutable: true,
      displayable: true,
      suspended: false,
      requestabet: false
    },
    boostCount: 0,
    superBoostCount: 0
  },
  {
    eventId: 21249937,
    name: 'Home United 0 v 1 Albirex Niigata FC',
    displayOrder: 100,
    sort: 'MTCH',
    linkedEventId: 21249693,
    classId: 5,
    className: 'Football',
    typeId: 10003971,
    typeName: 'Football Live',
    linkedEventTypeId: 10005913,
    linkedEventTypeName: 'Singapore S League',
    startTime: '2017-09-19T11:29:23.000Z',
    scores: { home: 0, away: 1 },
    competitors: [
      { name: 'Home United', position: 'home' },
      { name: 'Albirex Niigata FC', position: 'away' }
    ],
    status: {
      active: true,
      started: true,
      live: true,
      resulted: false,
      finished: false,
      cashoutable: true,
      displayable: true,
      suspended: false,
      requestabet: false
    },
    boostCount: 0,
    superBoostCount: 0
  }
]

const eventDetails = {
  eventId: 21249939,
  name: 'Shanghai Shenhua 0 v 0 Shandong Luneng Taishan',
  displayOrder: -1000,
  sort: 'MTCH',
  linkedEventId: 21228740,
  classId: 5,
  className: 'Football',
  typeId: 10003971,
  typeName: 'Football Live',
  linkedEventTypeId: 10005942,
  linkedEventTypeName: 'Chinese Super League',
  startTime: '2017-09-19T11:35:23.000Z',
  scores: { home: 0, away: 0 },
  competitors: [
    { name: 'Shanghai Shenhua', position: 'home' },
    { name: 'Shandong Luneng Taishan', position: 'away' }
  ],
  status: {
    active: true,
    started: true,
    live: true,
    resulted: false,
    finished: false,
    cashoutable: true,
    displayable: true,
    suspended: false,
    requestabet: false
  },
  boostCount: 0,
  superBoostCount: 0
}

const eventFullDetails = [
  {
    event: {
      eventId: 21249939,
      name: 'Shanghai Shenhua 0 v 0 Shandong Luneng Taishan',
      displayOrder: -1000,
      sort: 'MTCH',
      linkedEventId: 21228740,
      classId: 5,
      className: 'Football',
      typeId: 10003971,
      typeName: 'Football Live',
      linkedEventTypeId: 10005942,
      linkedEventTypeName: 'Chinese Super League',
      startTime: '2017-09-19T11:35:23.000Z',
      scores: { home: 0, away: 0 },
      competitors: [
        { name: 'Shanghai Shenhua', position: 'home' },
        { name: 'Shandong Luneng Taishan', position: 'away' }
      ],
      status: {
        active: true,
        started: true,
        live: true,
        resulted: false,
        finished: false,
        cashoutable: true,
        displayable: true,
        suspended: false,
        requestabet: false
      },
      boostCount: 0,
      superBoostCount: 0,
      markets: [
        93775335,
        93775336,
        93775337,
        93775338,
        93775339,
        93777094,
        93777375,
        93777385,
        93777503,
        93777682,
        93649785,
        93649830,
        93767581,
        93775329,
        93775330
      ]
    }
  },
  {
    event: {
      eventId: 21249939,
      name: 'Shanghai Shenhua 0 v 0 Shandong Luneng Taishan',
      displayOrder: -1000,
      sort: 'MTCH',
      linkedEventId: 21228740,
      classId: 5,
      className: 'Football',
      typeId: 10003971,
      typeName: 'Football Live',
      linkedEventTypeId: 10005942,
      linkedEventTypeName: 'Chinese Super League',
      startTime: '2017-09-19T11:35:23.000Z',
      scores: { home: 0, away: 0 },
      competitors: [
        { name: 'Shanghai Shenhua', position: 'home' },
        { name: 'Shandong Luneng Taishan', position: 'away' }
      ],
      status: {
        active: true,
        started: true,
        live: true,
        resulted: false,
        finished: false,
        cashoutable: true,
        displayable: true,
        suspended: false,
        requestabet: false
      },
      boostCount: 0,
      superBoostCount: 0,
      markets: [
        93775335,
        93775336,
        93775337,
        93775338,
        93775339,
        93777094,
        93777375,
        93777385,
        93777503,
        93777682,
        93649785,
        93649830,
        93767581,
        93775329,
        93775330
      ]
    }
  }
]

const betSlips = [
  {
    eventName: 'Kuching FA 0 v 1 UKM FC',
    marketName: 'Half-Time/Full-Time',
    name: 'Half-Time/Full-Time - Kuching FA/Kuching FA',
    outcome: {
      displayOrder: 10,
      eventId: 21249944,
      marketId: 93649434,
      name: 'Kuching FA/Kuching FA',
      outcomeId: 367529109,
      price: { decimal: '5.5', num: '9', den: '2' },
      result: { place: 0, result: '-', favourite: false },
      status: {
        active: true,
        resulted: false,
        cashoutable: false,
        displayable: true,
        suspended: false
      }
    }
  }
]

const currEventData = eventFullDetails.slice()[0]

// Using CommonJS style export so we can consume via Node (without using Babel-node)
module.exports = {
  sportEvents,
  eventFullDetails,
  eventDetails,
  betSlips
}
