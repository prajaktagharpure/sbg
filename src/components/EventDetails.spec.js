import React from 'react'
import { eventDetails } from '../../tools/mockData'
import EventDetails from './EventDetails'
import { shallow } from 'enzyme'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { create } from 'react-test-renderer'
import initialState from '../reducers/initialState'

describe('<EventDetails />', () => {
  const props = {
    eventDetails,
    currentEventData: { loading: true },
    actions: {
      handleAddBetSlip: jest.fn()
    }
  }

  it('should display loading text when full event data is loading', () => {
    const wrapper = shallow(<EventDetails {...props} />)
    expect(wrapper.find('.loadingtext').length).toBe(1)
  })

  it('should match snapshot', () => {
    const otherProps = {
      eventDetails,
      currentEventData: { loading: true },
      actions: {
        handleAddBetSlip: jest.fn()
      }
    }
    const store = configureMockStore()(initialState)
    const component = create(
      <Provider store={store}>
        <EventDetails {...otherProps} />
      </Provider>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
