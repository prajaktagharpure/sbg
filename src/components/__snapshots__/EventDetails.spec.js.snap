// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventDetails /> should match snapshot 1`] = `
<ul
  className="collection with-header event-details"
>
  <li
    className="collection-header"
  >
    <h4>
      Event Details
    </h4>
  </li>
  <li
    className="collection-item"
  >
    <strong>
      Event Name 
    </strong>
    Shanghai Shenhua 0 v 0 Shandong Luneng Taishan
  </li>
  <li
    className="collection-item"
  >
    <strong>
      Class Name 
    </strong>
    Football
  </li>
  <li
    className="collection-item"
  >
    <strong>
      Type Name
    </strong>
     
    Football Live
  </li>
  <li
    className="collection-item"
  >
    <strong>
      StartTime 
    </strong>
    2017-09-19T11:35:23.000Z
  </li>
  <li
    className="collection-item"
  >
    <strong>
       Home 
    </strong>
     
    0
     
    <br />
    <strong>
       Away 
    </strong>
    0
  </li>
  <li
    className="collection-item"
  >
    <strong>
      Competitors Home
    </strong>
     
    0
     
    <br />
    Shanghai Shenhua
    Shandong Luneng Taishan
  </li>
  <i
    className="center-align loadingtext"
  />
  <div
    className="progress indigo darken-4"
  >
    <div
      className="indeterminate"
    />
  </div>
</ul>
`;
